### ISSUE TITLE
[Feature Request] Add separator lines in Dockerfile for multistage build

### ISSUE THREAD

**Issue by ras0q** [microsoft/vscode-docker#4386]
If the extension could provide hints for separating each stage into sections, it would improve readability without the need for users to explicitly include comments separating the stages.

Slidev offers a feature similar to this.
https://sli.dev/features/vscode-extension

before|after
---|---
![image](https://github.com/user-attachments/assets/53515c9b-23e8-476e-9bde-41529996fd92) | ![image](https://github.com/user-attachments/assets/db1be488-8039-41aa-879b-a4a2c4cfce16)

**Comment by benwaffle** [comment]
Yeah, this would be nice. IntelliJ IDEs already do this:

<img width="679" alt="Image" src="https://github.com/user-attachments/assets/dd52c3ff-1038-4127-bd10-1b393b7cc357" />

**Comment by rcjsuen** [comment]
I'm not sure if the language server can do anything about this. @bwateratmsft Does VS Code have APIs to draw horizontal lines in the editor?

**Comment by bwateratmsft** [comment]
@rcjsuen We could maybe use CodeLens although that's not really the intended purpose. I think the closest thing would be [folding ranges](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_foldingRange), but that's already covered in issue #230.

I'll resolve this as a duplicate of #230. @ras0q @benwaffle if you think #230 is too different from what you're looking for, can you file an issue on VSCode to support horizontal lines?

**Comment by bwateratmsft** [comment]
Yeah, the LSP would be the right way to do this. Do you have any info on that custom drawing capability?

**Comment by benwaffle** [comment]
I know for certain you can do it in monaco, the editor library from vscode - https://microsoft.github.io/monaco-editor/playground.html?source=v0.52.2#example-interacting-with-the-editor-listening-to-mouse-events

So I assumed you can do the same thing via an extension

**Comment by rcjsuen** [comment]
> So I assumed you can do the same thing via an extension

Thanks for that example, @benwaffle. Looks like you can draw random lines with Monaco's `changeViewZones` API. I'm not familiar with the decorations API in VS Code so not sure if we can replicate the same thing here or not (even if we think outside the "language server box").

**Comment by bwateratmsft** [comment]
VSCode's extension APIs don't give direct access to the Monaco framework. There are webviews where you can do whatever you want, but the text editor isn't a webview in the sense that it cannot be extension-controlled, except through the APIs that VSCode offers.
